tests/unit/test_git_url_watcher.py

```diff
@@ -384,3 +491,65 @@ async def test_git_url_watcher_tags(
             assert latestTag == NEW_VALID_TAG_ON_NEW_SHA
     #
     await git_watcher.cleanup()
+
+
+async def test_git_url_watcher_tags_capture_group_replacement(
+    event_loop: AbstractEventLoop,
+    git_config_tags: dict[str, Any],
+):
+    local_path_var = git_config_tags["main"]["watched_git_repositories"][0][
+        "url"
+    ].replace("file://localhost", "")
+    repo_id_var = git_config_tags["main"]["watched_git_repositories"][0]["id"]
+    branch_var = git_config_tags["main"]["watched_git_repositories"][0]["branch"]
+
+    git_watcher = git_url_watcher.GitUrlWatcher(git_config_tags)
+
+    # the file does not exist yet
+    with pytest.raises(ConfigurationError):
+        init_result = await git_watcher.init()
+
+    # add the file
+    VALID_TAG = "teststaging_z1stvalid"
+    _run_cmd(
+        f"touch theonefile.csv; git add .; git commit -m 'I added theonefile.csv'; git tag {VALID_TAG};",
+        cwd=local_path_var,
+    )
+    await asyncio.sleep(0.5)
+    # expected to work now
+    init_result = await git_watcher.init()
+    git_sha = _run_cmd("git rev-parse --short HEAD", cwd=local_path_var)
+    assert init_result == {
+        repo_id_var: f"{repo_id_var}:{branch_var}:{VALID_TAG}:{git_sha}"
+    }
+
+    # there are no changes
+    assert not await git_watcher.check_for_changes()
+
+    NEW_VALID_TAG_ON_SAME_SHA: Literal[
+        "teststaging_z4thvalid"
+    ] = "teststaging_z4thvalid"
+    _run_cmd(
+        f"git tag {NEW_VALID_TAG_ON_SAME_SHA};",
+        cwd=local_path_var,
+    )
+    await asyncio.sleep(0.5)
+    NEW_VALID_TAG_ON_NEW_SHA: Literal[
+        "teststaging_h5thvalid"
+    ] = "teststaging_h5thvalid"  # This name is intentionally "in between" the previous tags when alphabetically sorted
+    _run_cmd(
+        f"echo 'blahblah' >> theonefile.csv; git add .; git commit -m 'I modified theonefile.csv'; git tag {NEW_VALID_TAG_ON_NEW_SHA}",
+        cwd=local_path_var,
+    )
+    await asyncio.sleep(1.5)
+    time.sleep(1.1)
+    # we should have a change here
```

Comment:
These sleeps look wired here.
TIP: You could take the opportunity to use some tenacity retry pattern in this test and make it more reliable.

Reply:
added a more decriptive function with reasoning why this is used, thanks